imag-todo/src/main.rs

```diff
@@ -16,210 +16,139 @@ extern crate libimagtodo;
-                                        Err(e) => {
-                                            trace_error(&e);
-                                            error!("{}", e);
-                                            return;
-                                        }
-                                    }
-                                }
-                                _ => {
-                                }
-                            } // end match ttask.status()
-                        } // end if c % 2
-                        counter += 1;
-                    } // end for
-                } // end if let
-                else {
-                    error!("No usable input");
-                }
-            }
-            else {
-                // Should not be possible, as one argument is required via
-                // ArgGroup
-                unreachable!();
-            }
-        },
-        Some("exec") => {
-            let subcmd = rt.cli().subcommand_matches("exec").unwrap();
-            let mut args = Vec::new();
-            if let Some(exec_string) = subcmd.values_of("command") {
-                for e in exec_string {
-                    args.push(e);
-                }
-                let tw_process = Command::new("task").stdin(Stdio::null()).args(&args).spawn().unwrap_or_else(|e| {
-                    panic!("failed to execute taskwarrior: {}", e);
-                });
-
-                let output = tw_process.wait_with_output().unwrap_or_else(|e| {
-                    panic!("failed to unwrap output: {}", e);
-                });
-                let outstring = String::from_utf8(output.stdout).unwrap_or_else(|e| {
-                    panic!("failed to ececute: {}", e);
-                });
-                println!("{}", outstring);
-            } else {
-                panic!("faild to execute: You need to exec --command");
-            }
-        }
-        Some("list") => {
-            let subcmd = rt.cli().subcommand_matches("list").unwrap();
-            let mut args = Vec::new();
-            let verbose = subcmd.is_present("verbose");
-            let iter = match libimagtodo::read::get_todo_iterator(rt.store()) {
-                //let iter = match rt.store().retrieve_for_module("todo/taskwarrior") {
-                Err(e) => {
-                    error!("{}", e);
-                    return;
-                }
-                Ok(val) => val,
-            };
-            for task in iter {
-                match task {
-                    Ok(val) => {
-                        //let val = libimagtodo::task::Task::new(fle);
-                        //println!("{:#?}", val.flentry);
-                        let uuid = match val.flentry.get_header().read("todo.uuid") {
-                            Ok(Some(u)) => u,
-                            Ok(None) => continue,
+                    }
+
+                    if *ttask.status() == TaskStatus::Deleted {
```

Comment:
A comment, why the check for `TaskStatus::Deleted` is absolutely necessary,  is missing.
```rust
// taskwarrior does not have hooks for deletion, but only for modifications,
// and of course we do not want to delete only modified tasks.
```